lib/spack/spack/environment/environment.py

```diff
@@ -883,9 +883,59 @@ def _process_definition(self, item):
             else:
                 self.spec_lists[name] = user_specs
 
+    def _process_view(self, env_view: Optional[Union[bool, str, Dict]]):
+        """Process view option(s), which can be boolean, string, or None.
+
+        A boolean environment view option takes precedence over any that may
+        be included. So ``view: True`` results in the default view only. And
+        ``view: False`` means the environment will have no view.
+
+        Args:
+            env_view: view option provided in the manifest or configuration
+        """
+
+        def add_view(name, values):
+            """Add the view with the name and the string or dict values IF
+            there is not already a view with that name."""
+            if name not in self.views:
+                if isinstance(values, str):
+                    self.views[name] = ViewDescriptor(self.path, values)
+                elif isinstance(values, dict):
+                    self.views[name] = ViewDescriptor.from_dict(self.path, values)
+                else:
+                    tty.error(f"Cannot add view named {name} for {type(values)} values {values}")
+            else:
+                tty.debug(f"Already have '{name}' view ({self.views[name]}), skipping {values}")
```

Comment:
This should be impossible to reach

Reply:
Good point.  Let me revisit that in light of #42058.

Reply:
Done.